src/transaction/log_manager.c

```diff
@@ -5255,18 +5292,39 @@ log_complete (THREAD_ENTRY * thread_p, LOG_TDES * tdes, LOG_RECTYPE iscommitted,
        */
       if (wrote_eot_log == LOG_NEED_TO_WRITE_EOT_LOG)
 	{
+	  // take care of calls that want to append commit only after commit_with_postpone
 	  if (iscommitted == LOG_COMMIT)
 	    {
 	      LOG_LSA commit_lsa;
 
-	      log_append_commit_log (thread_p, tdes, &commit_lsa);
+	      if (!LSA_ISNULL (&tdes->posp_nxlsa))
+		{
+		  log_append_finish_postpone (thread_p, tdes, &commit_lsa);
+		}
+	      else
+		{
+		  tx_group group;
+		  group.add (tdes->tran_index, 0, TRAN_UNACTIVE_COMMITTED);
+		  log_append_group_commit (thread_p, tdes, 0, group, &commit_lsa);
+		}
+
 	      log_change_tran_as_completed (thread_p, tdes, LOG_COMMIT, &commit_lsa);
 	    }
 	  else
 	    {
 	      LOG_LSA abort_lsa;
 
-	      log_append_abort_log (thread_p, tdes, &abort_lsa);
+	      if (!LSA_ISNULL (&tdes->posp_nxlsa))
+		{
+		  log_append_finish_postpone (thread_p, tdes, &abort_lsa);
+		}
+	      else
+		{
+		  tx_group group;
+		  group.add (tdes->tran_index, 0, TRAN_UNACTIVE_ABORTED);
+		  log_append_group_commit (thread_p, tdes, 0, group, &abort_lsa);
```

Comment:
I'd like to propose to rename `log_append_group_commit` as `log_append_group_complete`.
`commit` misleads me that transactions will be committed :)

There should be many consequences like `LOG_GROUP_COMMIT`, `LOG_REC_GROUP_COMMIT` and so on.

Reply:
done